demo/src/main/java/com/paypal/messagesdemo/XmlActivity.kt

```diff
@@ -29,23 +30,181 @@ class XmlActivity : AppCompatActivity() {
 
 		val payPalMessage = binding.payPalMessage
 		val progressBar = binding.progressBar
-		val reloadButton = binding.reloadButton
 
-		// TODO add example of adding MessageView here instead of in XML
+		val editedClientId: EditText? = findViewById<EditText>(R.id.clientId)
+
+		val logoTypeRadioGroup = findViewById<RadioGroup>(R.id.logoTypeRadioGroup)
+		logoTypeRadioGroup.setOnCheckedChangeListener { _, checkedId ->
+			when (checkedId) {
+				R.id.stylePrimary -> {
+					logoType = PayPalMessageLogoType.PRIMARY
+				}
+				R.id.styleInline -> {
+					logoType = PayPalMessageLogoType.INLINE
+				}
+				R.id.styleAlternative -> {
+					logoType = PayPalMessageLogoType.ALTERNATIVE
+				}
+				R.id.styleNone -> {
+					logoType = PayPalMessageLogoType.NONE
+				}
+			}
+		}
+
+		val colorRadioGroup = findViewById<RadioGroup>(R.id.colorRadioGroup)
+		colorRadioGroup.setOnCheckedChangeListener { _, checkedId ->
+			when (checkedId) {
+				R.id.styleBlack -> {
+					color = PayPalMessageColor.BLACK
+				}
+				R.id.styleWhite -> {
+					color = PayPalMessageColor.WHITE
+				}
+				R.id.styleMonochrome -> {
+					color = PayPalMessageColor.MONOCHROME
+				}
+				R.id.styleGrayscale -> {
+					color = PayPalMessageColor.GRAYSCALE
+				}
+			}
+		}
+
+		val alignmentRadioGroup = findViewById<RadioGroup>(R.id.alignmentRadioGroup)
+		alignmentRadioGroup.setOnCheckedChangeListener { _, checkedId ->
+			when (checkedId) {
+				R.id.styleLeft -> {
+					alignment = PayPalMessageAlign.LEFT
+				}
+				R.id.styleCenter -> {
+					alignment = PayPalMessageAlign.CENTER
+				}
+				R.id.styleRight -> {
+					alignment = PayPalMessageAlign.RIGHT
+				}
+			}
+		}
+
+		val shortTerm = findViewById<ToggleButton>(R.id.shortTerm)
+		val longTerm = findViewById<ToggleButton>(R.id.longTerm)
+		val payIn1 = findViewById<ToggleButton>(R.id.payIn1)
+		val credit = findViewById<ToggleButton>(R.id.credit)
+
+		fun updateOfferUi (offerName: String, isChecked: Boolean) {
```

Comment:
Why pass in a string instead of the offerType?

Reply:
I'll change it. I realized it needed to be a toggle instead of a radio input and switch everything at the last second and didn't really think about that section